.gitignore

```diff
@@ -14,10 +14,13 @@ npm-debug.log
 /lib/*
 !/lib/.npmignore
 
+/authProviders/*
```

Comment:
I'm not sure why this path is excluded when it contains MD files?

Reply:
authProviderOptions path contains MD files

Reply:
not clear, why ignore the md files?

Reply:
I am not  clear about this comment.

`authProviderOptions/ ` folder contains the md  file and the index.ts barrels for the submodules.  The `authProviderOptions/ `folder is not git ignored.
This folder is transpiled into the` authProviders/` folder which will be packaged in the npm.
`authProviders/` is ignored in git because it contains the transpiled x.d.ts and  x.js files. 
`authProviders/` is listed in the `package.json` files config so that it is bundled in the JS SDK npm.

Reply:
oooh ok, I got confused between the two, sorry about that and thanks for detailing!

Reply:
nit: conventions are usually to put everything that's going to be in the npm package but not on the repo in a lib or dist folder

Reply:
Yes. This approach that is used because of the limitations listed in the description.  This can be updated once the new 'exports' field in the package.json -> https://github.com/jkrems/proposal-pkg-exports/  is supported or typescript has a defined of doing this.

Reply:
what about putting it under lib/dist but without re-exporting it in the main barrel? This way we get a submodule (and it's msal/az-id deps) that's not necessarily in the tree at runtime (only if consumers explicitly take a dependency on it), we're aligned with conventions (making it work with older versions of typescript, not being dependent on a spec that's not implemented yet) and the pipeline/tooling configuration is simpler.

Reply:
The source is transpiled into the lib directory. We are not changing that. A user can import in both ways - 
1. `const {TokenCredentialAuthenticationProvider} = require("@microsoft/microsoft-graph-client/lib/src/authentication/azureTokenCredentials/TokenCredentialAuthenticationProvider");`
2. `const {TokenCredentialAuthenticationProvider} = require("@microsoft/microsoft-graph-client/authProvider/azureTokenCredentials/");`

Both work.
I considered building into the submodule in the lib folder, but I didn't see much difference between the two paths. 

I don't think there is a strict convention that everything should be inside the lib folder.

Some examples where multiple barrels are used to export in a submodule structure.
- https://github.com/facebook/react/tree/master/packages/react-dom - Example - react-dom/server
- https://github.com/wpengine/headless-framework/tree/main/packages/headless

Reply:
yeah, because we don't want barrels (to avoid taking hard dependencies) the lib/src part of the first "way of referencing it" cannot be shortened by convention and following the main directive in the package.json... I see.
Should we consider putting the auth provider in a separate package all together? This would resolve:
- keeping this package slim
- having the id packages as runtime dependencies
- having a clear import path
The only drawback is that people will need to install a second package, but that's not a big problem in this ecosystem as long as we document it.

Reply:
This point has been discussed and there is no substantial scope or functionality for a separate auth library for JS. 
If we are dependent on multiple authentication libraries a separate library would be a feasible option. With TokenCredentials, the goal would be to provide a single easy interface for a variety of authentication flows and keep the customAuthProvider implementation small.

Reply:
alright. Thanks for taking the time to discuss this. We'll see what feedback we get from the community once we ship :)